Change the windowize resource paths to something more linux friendly

--- ../../c30_resource/src/c30/c30_flag_definitions.h.~1~	2012-03-03 16:55:32.000000000 +0100
+++ ../../c30_resource/src/c30/c30_flag_definitions.h	2012-03-03 17:00:22.000000000 +0100
@@ -75,174 +75,131 @@
 #define DIR_SEPARATOR_STR "/"
 #endif
 
+#define	stringify1(x) #x
+#define	stringify(x) stringify1(x)
+#define	BASEDIR		stringify(PIC30PATH)
+#define	MBASEDIR	stringify(MCH_PIC30PATH)
+
 /*
  * assembly include paths for each device family
  *   Multiple paths may be separated by colons
  *   '/' is converted to the correct directory separator for the OS
  */
-#define MPLABC30_PIC30F_INC_PATH ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "dsPIC30F" DIR_SEPARATOR_STR \
-                                 "inc"
-#define MPLABC30_PIC33E_INC_PATH ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "dsPIC33E" DIR_SEPARATOR_STR \
-                                 "inc"
-#define MPLABC30_PIC33F_INC_PATH ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "dsPIC33F" DIR_SEPARATOR_STR \
-                                 "inc"
-#define MPLABC30_PIC33E_INC_PATH ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "dsPIC33E" DIR_SEPARATOR_STR \
-                                 "inc"
-#define MPLABC30_PIC24H_INC_PATH ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "PIC24H" DIR_SEPARATOR_STR \
-                                 "inc"
-#define MPLABC30_PIC24E_INC_PATH ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "PIC24E" DIR_SEPARATOR_STR \
-                                 "inc"
-#define MPLABC30_PIC24F_INC_PATH ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "PIC24F" DIR_SEPARATOR_STR \
-                                 "inc"
-#define MPLABC30_PIC24E_INC_PATH ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "PIC24E" DIR_SEPARATOR_STR \
-                                 "inc"
+#define MPLABC30_PIC30F_INC_PATH BASEDIR "/inc/dsPIC30F:" \
+	 			 MBASEDIR "/support/dsPIC30F/inc"
+
+#define MPLABC30_PIC33E_INC_PATH BASEDIR "/inc/dsPIC33E:" \
+	 			 MBASEDIR "/support/dsPIC33E/inc"
+
+#define MPLABC30_PIC33F_INC_PATH BASEDIR "/inc/dsPIC33F:" \
+	 			 MBASEDIR "/support/dsPIC33F/inc"
+
+#define MPLABC30_PIC24H_INC_PATH BASEDIR "/inc/PIC24H:" \
+	 			 MBASEDIR "/support/PIC24H/inc"
+
+#define MPLABC30_PIC24E_INC_PATH BASEDIR "/inc/PIC24E:" \
+	 			 MBASEDIR "/support/PIC24E/inc"
+
+#define MPLABC30_PIC24F_INC_PATH BASEDIR "/inc/PIC24F:" \
+	 			  MBASEDIR "/support/PIC24F/inc"
+
+
 /*
  * compiler include paths for each device family
  *   Multiple paths may be separated by colons
  *   '/' is converted to the correct directory separator for the OS
  */
-#define MPLABC30_COMMON_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
-                                     "include" PATH_SEPARATOR_STR \
-                                     ".." DIR_SEPARATOR_STR \
-                                     "support" DIR_SEPARATOR_STR \
-                                     "generic" DIR_SEPARATOR_STR \
-                                     "h"
-#define MPLABC30_LEGACY_COMMON_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
-                                     "include" DIR_SEPARATOR_STR "lega-c" \
-                                     PATH_SEPARATOR_STR \
-                                     ".." DIR_SEPARATOR_STR \
-                                     "support" DIR_SEPARATOR_STR \
-                                     "generic" DIR_SEPARATOR_STR \
-                                     "h"
-#define MPLABC30_PIC24F_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
-                                     "support" DIR_SEPARATOR_STR \
-                                     "PIC24F" DIR_SEPARATOR_STR \
-                                     "h" PATH_SEPARATOR_STR \
-                                     ".." DIR_SEPARATOR_STR \
-                                     "support" DIR_SEPARATOR_STR \
-                                     "peripheral_24F"
-#define MPLABC30_PIC24H_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
-                                     "support" DIR_SEPARATOR_STR \
-                                     "PIC24H" DIR_SEPARATOR_STR \
-                                     "h" PATH_SEPARATOR_STR \
-                                     ".." DIR_SEPARATOR_STR \
-                                     "support" DIR_SEPARATOR_STR \
-                                     "peripheral_30F_24H_33F"
-#define MPLABC30_PIC24E_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
-                                     "support" DIR_SEPARATOR_STR \
-                                     "PIC24E" DIR_SEPARATOR_STR \
-                                     "h" PATH_SEPARATOR_STR \
-                                     ".." DIR_SEPARATOR_STR \
-                                     "support" DIR_SEPARATOR_STR \
-                                     "peripheral_30F_24H_33F"
-#define MPLABC30_PIC30F_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
-                                     "support" DIR_SEPARATOR_STR \
-                                     "dsPIC30F" DIR_SEPARATOR_STR \
-                                     "h" PATH_SEPARATOR_STR \
-                                     ".." DIR_SEPARATOR_STR \
-                                     "support" DIR_SEPARATOR_STR \
-                                     "peripheral_30F_24H_33F"
-#define MPLABC30_PIC33F_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
-                                     "support" DIR_SEPARATOR_STR \
-                                     "dsPIC33F" DIR_SEPARATOR_STR \
-                                     "h" PATH_SEPARATOR_STR \
-                                     ".." DIR_SEPARATOR_STR \
-                                     "support" DIR_SEPARATOR_STR \
-                                     "peripheral_30F_24H_33F"
-#define MPLABC30_PIC33E_INCLUDE_PATH ".." DIR_SEPARATOR_STR \
-                                     "support" DIR_SEPARATOR_STR \
-                                     "dsPIC33E" DIR_SEPARATOR_STR \
-                                     "h" PATH_SEPARATOR_STR \
-                                     ".." DIR_SEPARATOR_STR \
-                                     "support" DIR_SEPARATOR_STR \
-                                     "peripheral_30F_24H_33F"
+#define MPLABC30_COMMON_INCLUDE_PATH	BASEDIR "/include:" \
+	 				BASEDIR "/include/generic:" \
+					MBASEDIR "/include/:" \
+					MBASEDIR "/support/generic/h"
+
+#define MPLABC30_LEGACY_COMMON_INCLUDE_PATH BASEDIR "/include/lega-c:" \
+					    BASEDIR "/include/generic:" \
+					    MBASEDIR "/include/lega-c:" \
+					    MBASEDIR "/support/generic/h"
+
+#define MPLABC30_PIC24F_INCLUDE_PATH	BASEDIR "/include/PIC24F:" \
+	 				BASEDIR "/include/peripheral_24F:" \
+					MBASEDIR "/support/PIC24F/h:" \
+					MBASEDIR "/support/peripheral_24F"
+
+#define MPLABC30_PIC24H_INCLUDE_PATH	BASEDIR "/include/PIC24H:" \
+	 				BASEDIR "/include/peripheral_30F_24H_33F:" \
+					MBASEDIR "/support/PIC24H/h:" \
+					MBASEDIR "/support/peripheral_30F_24H_33F"
+
+#define MPLABC30_PIC24E_INCLUDE_PATH	BASEDIR "/include/PIC24E:" \
+	 				BASEDIR "/include/peripheral_30F_24H_33F:" \
+					MBASEDIR "/support/PIC24E/h:" \
+					MBASEDIR "/support/peripheral_30F_24H_33F"
+
+#define MPLABC30_PIC30F_INCLUDE_PATH	BASEDIR "/include/dsPIC30F:" \
+	 				BASEDIR "/include/peripheral_30F_24H_33F:" \
+					MBASEDIR "/support/dsPIC30F/h:" \
+					MBASEDIR "/support/peripheral_30F_24H_33F"
+
+#define MPLABC30_PIC33F_INCLUDE_PATH	BASEDIR "/include/dsPIC33F:" \
+	 				BASEDIR "/include/peripheral_30F_24H_33F:" \
+					MBASEDIR "/support/dsPIC33F/h:" \
+					MBASEDIR "/support/peripheral_30F_24H_33F"
+
+#define MPLABC30_PIC33E_INCLUDE_PATH	BASEDIR "/include/dsPIC33E:" \
+					BASEDIR "/include/peripheral_30F_24H_33F:" \
+					MBASEDIR "/support/dsPIC33E/h:" \
+					MBASEDIR "/support/peripheral_30F_24H_33F"
+
 /*
  * linker library paths for each device family
  *   Multiple paths may be separated by colons
  *   '/' is converted to the correct directory separator for the OS
  */
-#define MPLABC30_COMMON_LIB_PATH ".." DIR_SEPARATOR_STR \
-                                 "lib" PATH_SEPARATOR_STR \
-                                 ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "generic" DIR_SEPARATOR_STR \
-                                 "gld"
-#define MPLABC30_PIC24F_LIB_PATH ".." DIR_SEPARATOR_STR \
-                                 "lib" DIR_SEPARATOR_STR \
-                                 "PIC24F" PATH_SEPARATOR_STR \
-                                 ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "PIC24F" DIR_SEPARATOR_STR \
-                                 "gld" PATH_SEPARATOR_STR \
-                                 ".." DIR_SEPARATOR_STR \
-                                 "lib" DIR_SEPARATOR_STR \
-                                 "peripheral_24F"
-#define MPLABC30_PIC24H_LIB_PATH ".." DIR_SEPARATOR_STR \
-                                 "lib" DIR_SEPARATOR_STR \
-                                 "PIC24H" PATH_SEPARATOR_STR \
-                                 ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "PIC24H" DIR_SEPARATOR_STR \
-                                 "gld" PATH_SEPARATOR_STR \
-                                 ".." DIR_SEPARATOR_STR \
-                                 "lib" DIR_SEPARATOR_STR \
-                                 "peripheral_30F_24H_33F"
-#define MPLABC30_PIC24E_LIB_PATH ".." DIR_SEPARATOR_STR \
-                                 "lib" DIR_SEPARATOR_STR \
-                                 "PIC24E" PATH_SEPARATOR_STR \
-                                 ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "PIC24E" DIR_SEPARATOR_STR \
-                                 "gld" PATH_SEPARATOR_STR \
-                                 ".." DIR_SEPARATOR_STR \
-                                 "lib" DIR_SEPARATOR_STR \
-                                 "peripheral_30F_24H_33F"
-#define MPLABC30_PIC30F_LIB_PATH ".." DIR_SEPARATOR_STR \
-                                 "lib" DIR_SEPARATOR_STR \
-                                 "dsPIC30F" PATH_SEPARATOR_STR \
-                                 ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "dsPIC30F" DIR_SEPARATOR_STR \
-                                 "gld" PATH_SEPARATOR_STR \
-                                 ".." DIR_SEPARATOR_STR \
-                                 "lib" DIR_SEPARATOR_STR \
-                                 "peripheral_30F_24H_33F"
-#define MPLABC30_PIC33F_LIB_PATH ".." DIR_SEPARATOR_STR \
-                                 "lib" DIR_SEPARATOR_STR \
-                                 "dsPIC33F" PATH_SEPARATOR_STR \
-                                 ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "dsPIC33F" DIR_SEPARATOR_STR \
-                                 "gld" PATH_SEPARATOR_STR \
-                                 ".." DIR_SEPARATOR_STR \
-                                 "lib" DIR_SEPARATOR_STR \
-                                 "peripheral_30F_24H_33F"
-#define MPLABC30_PIC33E_LIB_PATH ".." DIR_SEPARATOR_STR \
-                                 "lib" DIR_SEPARATOR_STR \
-                                 "dsPIC33E" PATH_SEPARATOR_STR \
-                                 ".." DIR_SEPARATOR_STR \
-                                 "support" DIR_SEPARATOR_STR \
-                                 "dsPIC33E" DIR_SEPARATOR_STR \
-                                 "gld" PATH_SEPARATOR_STR \
-                                 ".." DIR_SEPARATOR_STR \
-                                 "lib" DIR_SEPARATOR_STR \
-                                 "peripheral_30F_24H_33F"
 
+#define MPLABC30_COMMON_LIB_PATH	BASEDIR "/lib:" \
+	 				BASEDIR "/gld:" \
+					MBASEDIR "/lib:" \
+					MBASEDIR "/support/generic/gld"
+
+#define MPLABC30_PIC24F_LIB_PATH     	BASEDIR "/lib/PIC24F:" \
+	 			      	BASEDIR "/gld/PIC24F:" \
+				      	BASEDIR "/lib/peripheral_24F:" \
+				      	MBASEDIR "/lib/PIC24F:" \
+				      	MBASEDIR "/support/PIC24F/gld:" \
+				      	MBASEDIR "/lib/peripheral_24F"
+
+#define MPLABC30_PIC24H_LIB_PATH	BASEDIR "/lib/PIC24H:" \
+				      	BASEDIR "/gld/PIC24H/:" \
+				      	BASEDIR "/lib/peripheral_30F_24H_33F:" \
+				      	MBASEDIR "/lib/PIC24H:" \
+				      	MBASEDIR "/support/PIC24H/gld:" \
+				      	MBASEDIR "/lib/peripheral_30F_24H_33FF"
+
+#define MPLABC30_PIC24E_LIB_PATH     	BASEDIR "/lib/PIC24E:" \
+				      	BASEDIR "/gld/PIC24E:" \
+				      	BASEDIR "/lib/peripheral_30F_24H_33F:" \
+				      	MBASEDIR "/lib/PIC24E:" \
+				      	MBASEDIR "/support/PIC24E/gld:" \
+				      	MBASEDIR "/lib/peripheral_30F_24H_33FF"
+
+#define MPLABC30_PIC30F_LIB_PATH     	BASEDIR "/lib/dsPIC30F:" \
+				      	BASEDIR "/gld/dsPIC30F:" \
+				      	BASEDIR "/lib/peripheral_30F_24H_33F:" \
+				      	MBASEDIR "/lib/dsPIC30F:" \
+				      	MBASEDIR "/support/dsPIC30F/gld:" \
+				      	MBASEDIR "/lib/peripheral_30F_24H_33FF"
+
+#define MPLABC30_PIC33F_LIB_PATH     	BASEDIR "/lib/dsPIC33F:" \
+				      	BASEDIR "/gld/dsPIC33F:" \
+				      	BASEDIR "/lib/peripheral_30F_24H_33F:" \
+				      	MBASEDIR "/lib/dsPIC33F:" \
+				      	MBASEDIR "/support/dsPIC33F/gld:" \
+				      	MBASEDIR "/lib/peripheral_30F_24H_33FF"
+ 
+#define MPLABC30_PIC33E_LIB_PATH     	BASEDIR "/lib/dsPIC33E:" \
+				      	BASEDIR "/gld/dsPIC33E:" \
+				      	BASEDIR "/lib/peripheral_30F_24H_33F:" \
+				      	MBASEDIR "/lib/dsPIC33E:" \
+				      	MBASEDIR "/support/dsPIC33E/gld:" \
+				      	MBASEDIR "/lib/peripheral_30F_24H_33FF"
 
 #endif
